File: catalog/backends/ruby-rails/patterns/unsafe_search_controller.rb
# Pattern: Code Security — SQL injection via string interpolation
# WARNING: intentionally vulnerable for IAST demo
# Adapt: replace table/column with domain entity

def search
  q = params[:q].to_s
  sql = "SELECT * FROM items WHERE name LIKE '%#{q}%'" # rubocop:disable Rails/SquishedSQLHeredocs
  results = ActiveRecord::Base.connection.execute(sql).to_a
  render json: results
end

File: catalog/backends/php-laravel/patterns/UnsafeSearchController.php
<?php

use Illuminate\Support\Facades\DB;
use Illuminate\Http\Request;
use Illuminate\Http\JsonResponse;

// Pattern: Code Security — SQL injection via string concat
// WARNING: intentionally vulnerable for IAST demo

function search(Request $request): JsonResponse
{
    $q = $request->query('q', '');
    $rows = DB::select("SELECT * FROM items WHERE name LIKE '%" . $q . "%'");
    return response()->json($rows);
}
